File: app/src/main/java/com/example/food_app/activity/ListFoodActivity.java
package com.example.food_app.activity;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.food_app.adapter.FoodListAdapter;
import com.example.food_app.databinding.ActivityListBinding;
import com.example.food_app.domain.Foods;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.Query;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;

public class ListFoodActivity extends BaseActivity {
    ActivityListBinding binding;
    private int categoryId;
    private String categoryName;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityListBinding.inflate(getLayoutInflater());

        setContentView(binding.getRoot());

        getIntentExtra();
        initList();
    }

    private void initList() {
        DatabaseReference ref = database.getReference("Foods");
        binding.progressBar.setVisibility(View.VISIBLE);
        ArrayList<Foods> list = new ArrayList<>();
        Query query = ref.orderByChild("CategoryId").equalTo(categoryId);

        query.addListenerForSingleValueEvent(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot snapshot) {
                if (snapshot.exists()) {
                    for (DataSnapshot issue : snapshot.getChildren()) {
                        list.add(issue.getValue(Foods.class));
                    }
                    if (!list.isEmpty()) {
                        binding.foodListView.setLayoutManager(new LinearLayoutManager(ListFoodActivity.this, LinearLayoutManager.VERTICAL, false));
                        binding.foodListView.setAdapter(new FoodListAdapter(list));
                    }
                    binding.progressBar.setVisibility(View.GONE);
                }
            }

            @Override
            public void onCancelled(@NonNull DatabaseError error) {

            }
        });
    }

    private void getIntentExtra() {
        categoryId = getIntent().getIntExtra("CategoryId", 0);
        categoryName = getIntent().getStringExtra("CategoryName");

        binding.titleTxt.setText(categoryName);
        binding.backButton.setOnClickListener(v -> finish());
    }
}

File: app/src/main/java/com/example/food_app/activity/ListFoodActivity.kt
package com.example.food_app.activity

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.food_app.adapter.FoodListAdapter
import com.example.food_app.databinding.ActivityListBinding
import com.example.food_app.domain.Foods
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.ValueEventListener

class ListFoodActivity : BaseActivity() {
    var binding: ActivityListBinding? = null
    private var categoryId = 0
    private var categoryName: String? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityListBinding.inflate(layoutInflater)

        setContentView(binding!!.root)

        intentExtra
        initList()
    }

    private fun initList() {
        val ref = database!!.getReference("Foods")
        binding!!.progressBar.visibility = View.VISIBLE
        val list = ArrayList<Foods?>()
        val query = ref.orderByChild("CategoryId").equalTo(categoryId.toDouble())

        query.addListenerForSingleValueEvent(object : ValueEventListener {
            override fun onDataChange(snapshot: DataSnapshot) {
                if (snapshot.exists()) {
                    for (issue in snapshot.children) {
                        list.add(issue.getValue(Foods::class.java))
                    }
                    if (!list.isEmpty()) {
                        binding!!.foodListView.layoutManager = LinearLayoutManager(
                            this@ListFoodActivity,
                            LinearLayoutManager.VERTICAL,
                            false
                        )
                        binding!!.foodListView.adapter = FoodListAdapter(list)
                    }
                    binding!!.progressBar.visibility = View.GONE
                }
            }

            override fun onCancelled(error: DatabaseError) {
            }
        })
    }

    private val intentExtra: Unit
        get() {
            categoryId = intent.getIntExtra("CategoryId", 0)
            categoryName = intent.getStringExtra("CategoryName")

            binding!!.titleTxt.text = categoryName
            binding!!.backButton.setOnClickListener { v -> finish() }
        }
}